downstairs/src/extent_inner_raw_v2.rs

```diff
@@ -0,0 +1,1013 @@
+// Copyright 2023 Oxide Computer Company
+use crate::{
+    cdt,
+    extent::{check_input, extent_path, ExtentInner, EXTENT_META_RAW_V2},
+    extent_inner_raw_common::{
+        pread_all, pwrite_all, OnDiskMeta, BLOCK_META_SIZE_BYTES,
+    },
+    mkdir_for_file,
+    region::JobOrReconciliationId,
+    Block, CrucibleError, ExtentReadRequest, ExtentReadResponse, ExtentWrite,
+    JobId, RegionDefinition,
+};
+
+use crucible_common::{BlockOffset, ExtentId};
+use crucible_protocol::ReadBlockContext;
+use slog::{error, Logger};
+
+use std::io::{BufReader, Read};
+use std::os::fd::{AsFd, AsRawFd};
+use std::path::Path;
+use std::{
+    fs::{File, OpenOptions},
+    io::IoSlice,
+};
+use zerocopy::AsBytes;
+
+/// Recordsize for non-ZFS systems (matching the ZFS default)
+pub(crate) const DUMMY_RECORDSIZE: u64 = 128 * 1024;
+
+/// Max of serialized block context (which is a [`ReadBlockContext`])
+pub(crate) const BLOCK_CONTEXT_SIZE_BYTES: u64 = 32;
+
+/// `RawInnerV2` is a wrapper around a [`std::fs::File`] representing an extent
+///
+/// The file contains [`extent_size`](RawInnerV2::extent_size) blocks, which I'm
+/// writing as `N` for simplicity here.
+///
+/// # File organization
+/// The file is structured in three parts.  Getting specific offsets within the
```

Comment:
I would think this is 4 parts?
`[Block data and context]`
`[Expected recordsize]`
`[Written block array]`
`[File metadata]`

Ah, maybe `[Written block array]` and `[File metadata]` are considered the same part because
we write them in a single write?

Reply:
This was outdated, fixed!